app/mailers/user_mailer.rb

```diff
@@ -1,8 +1,15 @@
 class UserMailer < ApplicationMailer
-  def authentication
-    @authentication = params[:authentication]
-    @user = @authentication.user
+  before_action :set_user
 
-    mail to: @user.email_address, subject: "Sign in to Hackathons"
+  def authentication(authentication = nil)
+    @authentication = authentication || @user.authentications.last
+
+    mail to: @authentication.user.email_address, subject: "Sign in to Hackathons"
+  end
+
+  private
+
+  def set_user
+    @user = params & [:user] || Current.user
```

Comment:
Is this a typo?

Reply:
Nah, for some reason Standard requires a space between the ampersand and `params` isn't always defined iirc, though there's probably a better way to do that.

Reply:
Try this. i'm not sure if it's any better or if Standard will complain

```suggestion
    @user = params&.[](:user) || Current.user
```

Since `[]` is a method, it can also be called with parentheses

Reply:
wasn't sure if that looked better, but it's definitely less confusing!